packages/web-components/src/card/card.ts

```diff
@@ -0,0 +1,207 @@
+  @attr
+  public appearance?: CardAppearance;
+
+  /**
+   * @property orientation;
+   * @default vertical
+   * @remarks
+   * Determines the orientation of the card
+   */
+  @attr
+  public orientation?: CardOrientation;
+
+  /**
+   * @property controlSize
+   * @default medium
+   * @remarks
+   * Determines the size of the card
+   */
+  @attr({ attribute: 'control-size' })
+  public controlSize?: CardControlSize;
+
+  /**
+   * @property interactive
+   * @default false
+   * @remarks
+   * Determines whether card is interactable
+   */
+  @observable
+  @attr({ mode: 'boolean' })
+  public interactive: boolean = false;
+
+  /**
+   * @property disabled
+   * @default false
+   * @remarks
+   * Determines disabled state of card
+   */
+  @attr({ mode: 'boolean' })
+  public disabled: boolean = false;
+
+  /**
+   * @property selected
+   * @default false
+   * @remarks
+   * Determines selected state of card
+   */
+  @observable
+  @attr({ mode: 'boolean' })
+  public selected: boolean = false;
+
+  /**
+   * @method sizeChanged
+   * @remarks
+   * Updates the computed stylesheet when the size of the card changes
+   */
+  public sizeChanged(prev: boolean | undefined, next: boolean): void {
+    this.updateComputedStylesheet();
+  }
+
+  /**
+   * @method selectedChanged
+   * @remarks
+   * Emits an event when the selected state of the card changes
+   */
+  public selectedChanged = (): void => {
+    this.$emit('onSelectionChanged', this.selected);
+  };
+
+  /**
+   * @method interactiveChanged
+   * @remarks
+   * Updates the focusable state of the card when the interactive state changes
+   */
+  public interactiveChanged = (): void => {
+    this.setInteractive();
+  };
+
+  /**
+   * @method setFocusable
+   * @remarks
+   * Sets the tabindex attribute based on the interactive state of the card
+   * @internal
+   */
+  public setInteractive(): void {
+    if (this.interactive) {
+      this.setAttribute('tabindex', '0');
+      this.setAttribute('aria-selected', this.selected ? 'true' : 'false');
+    } else {
+      this.removeAttribute('tabindex');
+      this.removeAttribute('aria-selected');
+    }
+  }
+
+  /**
+   * Updates an internal stylesheet with calculated CSS custom properties.
+   *
+   * @internal
+   */
+  protected updateComputedStylesheet(): void {
+    // Determine the pixel value based on the controlSize attribute
+    let sizeValue;
+    let borderRadiusValue;
+    switch (this.controlSize) {
+      case CardControlSize.small:
+        sizeValue = '8px';
+        borderRadiusValue = '2px';
+        break;
+      case CardControlSize.medium:
+        sizeValue = '12px';
+        borderRadiusValue = '4px';
+        break;
+      case CardControlSize.large:
+        sizeValue = '16px';
+        borderRadiusValue = '8px';
+        break;
+      default:
+        sizeValue = '12px';
+        borderRadiusValue = '4px';
+        break;
+    }
+
+    this.$fastController.removeStyles(this.computedStylesheet);
```

Comment:
We might be able to reduce this a tad if we define the default values before the switch statement, e.g.
```ts
protected updateComputedStylesheet(): void {
  const defaultSize = '12px';
  const defaultBorderRadius = '4px';
  let sizeValue = defaultSize;
  let borderRadiusValue = defaultBorderRadius;

  switch (this.controlSize) {
    case CardControlSize.small:
      sizeValue = '8px';
      borderRadiusValue = '2px';
      break;
    case CardControlSize.medium:
      sizeValue = defaultSize;
      borderRadiusValue = defaultBorderRadius;
      break;
    case CardControlSize.large:
      sizeValue = '16px';
      borderRadiusValue = '8px';
      break;
  }
...
}
```

Reply:
I'm not sure I understand why this is needed - this feels like something which should either be spelled out in the spec, or commented. Can you help me understand what the intent of this is?